modules/generic-extras/src/main/scala/io/circe/generic/extras/Auto.scala

```diff
@@ -10,11 +10,11 @@ import scala.language.experimental.macros
 /**
  * Fully automatic configurable codec derivation.
  *
- * Importing the contents of this object provides [[io.circe.Decoder]] and [[io.circe.Encoder]]
+ * Extending this trait provides [[io.circe.Decoder]] and [[io.circe.Encoder]]
  * instances for case classes (if all members have instances), "incomplete" case classes, sealed
  * trait hierarchies, etc.
  */
-final object auto {
+trait Auto {
```

Comment:
What do you think about calling this `AutoDerivation` instead? I'd find it a bit clearer in cases where it's imported and then mixed into something like a protocol object that might have a number of other traits involved.